commands/scs_dice.py

```diff
@@ -351,4 +352,104 @@ def modify_rollcall(self, rollcall, args):
+        # many steps will be in the roll string. As I add more options, like "staking" buffs/debuffs on the roll result,
+        # I can modify max_steps based on self.switches so the player can specify how complex the roll will be.
+        step_counter = 0
+        max_steps = 3
+        dice_string_list = []
+        while step_counter <= max_steps:
+            if step_counter == -1:
+                # Response_parser will set step_counter to -1 if the user specifies to quit.
+                return caller.msg("Cancelling EZRoll.")
+            elif step_counter == 0:
+                caller.msg("EZRoll is preparing to roll dice step-by-step. ('u' to undo a step, 'q' to quit)")
+                response = yield("How many dice of how many sides would you like to roll? (e.g., 1d20)")
+            elif step_counter == 1:
+                response = yield("What modifier would you like to add? (e.g., +2, -1. 'enter' for none)")
+            elif step_counter == 2:
+                response = yield("What target number do you aim to exceed? (e.g., 10. 'enter' for none)")
+            elif step_counter == 3:
+                response = yield("Add an optional comment to describe the roll. ('enter' for none)")
+            step_counter, dice_string_list = self.response_parser(response, step_counter, dice_string_list)
+        # Once the while loop concludes, send the joined dice_string to CmdHandler and run SCSDice on it.
+        dice_string = "".join(dice_string_list)
+        # Calling cmdhandler directly feels sketchy, but the roll code is a contrib, so this seems like the simplest
+        # approach to integrating a new command, EZRoll, that is dependent on said roll code.
+        cmdhandler(caller, f"roll {dice_string}")
+
+    def response_parser(self, response, step_counter, dice_string_list):
+        # Parse and corner case depending on the step in assembling the dice roll
+
+        def success(resp, step_count, dice_string_l):
+            # Writing a sub-function to call when the response is well-formed to help with efficient corner-casing
+            step_count += 1
+            dice_string_l.append(resp)
+            return step_count, dice_string_l
+
+        caller = self.caller
+        # Check for undo or quit input
+        if response == "u":
+            # Go back a step and remove the most recent append to dice_string_list
+            step_counter -= 1
+            dice_string_list.pop()
+            caller.msg("Undoing previous step.")
+            return step_counter, dice_string_list
+        elif response == "q":
+            # Shortcut to break the while loop in the main function
+            step_counter = -1
+            return step_counter, dice_string_list
+        elif step_counter == 0:
+            if "d" not in response:
+                # There should be a "d" in the first dice roll specification
+                caller.msg("Invalid response. Please specify a valid dice roll, e.g., 1d20 or 3d6.")
```

Comment:
probably want more robust error handling here - there are a bunch of weird corner cases since people can type whatever the hell they want ("d20", "3d", "5d2.5", etc)

Reply:
the roll function itself definitely has this error handling built in, but it just depends on how convenient you want the functionality to be